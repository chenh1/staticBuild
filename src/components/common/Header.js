import React, {PropTypes} from 'react';
import { Link, IndexLink } from 'react-router';

class Header extends React.Component {
  render(){
    return (
      <nav id="header" className="header">
        <IndexLink to="/" id="logo" className="logo" activeClassName="active">
          <img src="../assets/logo.png" alt="Tracy Lin Logo" />
          <h1 className="page-name">
            Tracy Lin
          </h1>
        </IndexLink>
        
        <ul className="nav-menu">
          <li><Link to="/work" className="page-link" activeClassName="active">Work</Link></li>
          <li><Link to="/about" className="page-link" activeClassName="active">About</Link></li>
          <li><a href="/resume.pdf" className="page-link" activeClassName="active">Resume</a></li>
        </ul>
      </nav>
    );
  }
}

export default Header;
