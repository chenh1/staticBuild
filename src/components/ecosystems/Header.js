import React, {PropTypes} from 'react';
import { Link, IndexLink } from 'react-router';
import { BulletList } from '../molecules';

const Header = ({listItems}) => {
  return (
    <nav id="header" className="header">
      <IndexLink to="/" id="logo" className="logo" activeClassName="active">
        <img src={require("../../assets/logo.png")} alt="Tracy Lin Logo" />
        <h1 className="page-name">
          Tracy Lin
        </h1>
      </IndexLink>

      <BulletList
        styleClass="nav-menu"
        listItems={listItems}
      />
    </nav>
  );
}

export default Header;
