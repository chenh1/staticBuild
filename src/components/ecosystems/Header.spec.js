import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Header from './Header';

test('renders header', () => {
    const tree = renderer.create(<Header
        headerClasses="mock-class"
        headerLogo="/mockLogoSrc"
        headerNavItems= {[
            {path: "/work", linkClass:"page-link", activeClassName:"active", text:"Work"},
            {path: "/about", linkClass:"page-link", activeClassName:"active", text:"About"},
            {path: '/mockResumeUrl', linkClass:"page-link", activeClassName:"active", text:"Resume"}
        ]} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});