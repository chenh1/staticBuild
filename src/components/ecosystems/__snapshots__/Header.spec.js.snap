// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header 1`] = `
<nav
  className="mock-class"
  id="header"
>
  <a
    className="logo"
    id="logo"
    onClick={[Function]}
    style={Object {}}
  >
    <img
      alt="Tracy Lin Logo"
      src="/mockLogoSrc"
    />
    <h1
      className="page-name"
    >
      Tracy Lin
    </h1>
  </a>
  <ul
    className="nav-menu"
  >
    <li>
      <a
        className="page-link"
        onClick={[Function]}
        style={Object {}}
      >
        Work
      </a>
    </li>
    <li>
      <a
        className="page-link"
        onClick={[Function]}
        style={Object {}}
      >
        About
      </a>
    </li>
    <li>
      <a
        className="page-link"
        onClick={[Function]}
        style={Object {}}
      >
        Resume
      </a>
    </li>
  </ul>
</nav>
`;
